### PR TITLE
nvidia: support disabling the nvidia plugin

### PR DESCRIPTION
**Pull request by notnoop**
This provides a workaround for where nomad does not support the nvidia driver library installed on host.  The crash in https://github.com/hashicorp/nomad/issues/8303 is a result of an attempt to fingerprint using a C function that's not resolved, so process crashes.  Because this happens before agent is up, the logs are completely hidden.

To disable the plugin, one adds the following to the agent config:

```hcl
plugin "nvidia-gpu" {
  config {
    enabled = false
  }
}
```

Fixes https://github.com/hashicorp/nomad/issues/8303

### CODE REVIEW

File: devices/gpu/nvidia/device.go [hunk 1]
```diff
@@ -149,6 +159,10 @@ func (d *NvidiaDevice) SetConfig(cfg *base.Config) error {
 // Fingerprint streams detected devices. If device changes are detected or the
 // devices health changes, messages will be emitted.
 func (d *NvidiaDevice) Fingerprint(ctx context.Context) (<-chan *device.FingerprintResponse, error) {
+	if !d.enabled {
+		return nil, errDeviceNotEnabled
```
Comment (tgross) [inline devices/gpu/nvidia/device.go:163]: I'm not 100% clear on what triggers the fingerprint call... are we going to end up spamming the logs if we return an error here: https://github.com/hashicorp/nomad/blob/v0.12.0/client/devicemanager/instance.go#L340 or does this just happen once?

(And should it be logged as an error at all if we disabled it?)

Comment (notnoop) [reply devices/gpu/nvidia/device.go:163]: When fingerprinting, the device manager gives up and doesn't fingerprinting again.  So we only log it once.

It's logged as an error like the following:

```
    2020-07-21T13:27:43.558Z [ERROR] client.device_mgr: fingerprinting failed: plugin=nvidia-gpu error="Nvidia device is not enabled"
```

I can make it a debug/info statement.

Comment (notnoop) [reply devices/gpu/nvidia/device.go]: Made it info in 7896d1f8a9097936e03162ca0e73341a370f7d7a .

Comment (tgross) [reply devices/gpu/nvidia/device.go]: LGTM!

**Pull request merged**